core/variant/variant.h

```diff
@@ -441,12 +443,13 @@ class Variant {
 	operator Vector<Variant>() const;
 	operator Vector<StringName>() const;
 
-	// some core type enums to convert to
-	operator Side() const;
-	operator Orientation() const;
-
 	operator IPAddress() const;
 
+	template <typename T, std::enable_if_t<std::is_enum_v<T>, int> = 0>
+	_FORCE_INLINE_ operator T() const { return static_cast<T>(operator int64_t()); }
```

Comment:
Interesting generalized approach! Two comments:

1) While the source code doesn't currently differentiate anyway, we may want to use the appropriate `operator()` based on the actual enum type (`std::underlying_type_t<FooEnum>`).
2) This is opening a new can of worms for implicit conversions, where previously only `Side` and `Orientation` were allowed. Perhaps we can make this explicit, and re-add `Side` and `Orientation` back as implicit for backwards compat (for now)?

Reply:
1. All int-likes are cast to/from `int64_t` with Variant as-is, so that being the baseline should be fine (or, at least, working-as-expected).
2. For better or for worse, Variant is built upon implicit casts. Those two were specified before for convenience, in a manner camp to the non-packed `Vector<>` overloads; I wouldn't consider that *excluding* other enums, so much as not wanting to deal with extra verbosity. In other words: this won't break compat, as it's not changing any behavior for the existing code.

Reply:
1. If we accept this at the baseline, i think I should rewrite #101055 with the same logic (call `operator int64_t`, `cast`). Agree? I'd be fine with that too.

Reply:
You aren't dealing with generics in that PR, so it's probably not *as* necessary. If anything, I'd consider leaning *more* into the generics with that PR. Lemme setup a diff real quick; I'll post it on the PR itself to not derail this